### PR TITLE
Produce junit test report XML on Circle CI

### PR DESCRIPTION
**Pull request by mfurtak**
This enables Circle's web test report display functionality

https://circleci.com/docs/test-metadata#rspec

screen shot 2016-03-18 at 3 26 50 pm

screen shot 2016-03-19 at 12 59 16 pm

### CODE REVIEW

File: fastlane/spec/unused_options_spec.rb [hunk 1]
```diff
@@ -3,6 +3,15 @@
     describe "No unused options" do
       let (:all_exceptions) { %w(pilot appstore cert deliver gym match pem produce scan sigh snapshot supply testflight mailgun testfairy ipa import_from_git hockey deploygate crashlytics artifactory appledoc slather screengrab download_dsyms) }
 
+      # https://github.com/bundler/bundler/issues/4368
+      if Object.const_defined?(:Bundler) && Bundler.const_defined?(:SpecSet) && Bundler::SpecSet.instance_methods.include?(:length)
```
Comment (lacostej) [inline fastlane/spec/unused_options_spec.rb]: should probably add a
```&& ! Bundler::SpecSet.instance_methods.include?(:size)```

Comment (mfurtak) [inline fastlane/spec/unused_options_spec.rb]: Will do

**Comment by KrauseFx** [comment]
Where can I see this in Circle? It only shows the slowest test, right? Or is there more information available?

**Comment by mfurtak** [comment]
It sounds like you're looking at the right place. My suspicion is that it will show more detailed output if there are failing tests.

On early runs of this, I think this change was conflicting with Danger's expectations around test error reporting (Danger expects every file in the reports directory to be JSON).  We may need to fix that if we want to also have this.

**Comment by mfurtak** [comment]
Actually, I took another look at the `Dangerfile` and the current .json report file collecting seems just fine. @KrauseFx let me know if you think this is worth having :+1:

**Comment by KrauseFx** [comment]
So what advantage do we have by keeping the JUnit file? It seems like `circle` isn't really using it, right?

**Comment by mfurtak** [comment]
Here is what the Circle build report looks like with a failing test after these changes:

screen shot 2016-03-19 at 12 59 16 pm

**Comment by KrauseFx** [comment]
:rocket: yeah

**Pull request merged**